src/datasources/balances-api/entities/asset-price.entity.ts

```diff
@@ -1,3 +1,6 @@
-export interface AssetPrice {
-  [assetName: string]: Record<string, number | null>;
-}
+import { z } from 'zod';
+
+export type AssetPrice = z.infer<typeof AssetPriceSchema>;
+
+// TODO: Enforce Ethereum address keys (and maybe checksum them)
```

Comment:
I've left this note as it would tie in with refactoring the `CoingeckoApi`. As their API accepts checksummed addresses, we should only work with them - checksumming the response too.